src/db/graph/exec.rs

```diff
@@ -25,3 +26,24 @@ pub async fn exec_boolean_row(query: Query) -> Result<bool, DynError> {
     }
     Ok(boolean)
 }
+
+// Generic function to retrieve data from Neo4J
+pub async fn retrieve_from_graph<T>(query: Query, key: &str) -> Result<Option<T>, DynError>
```

Comment:
better in that file! like this we could use it with others that has the same behavior. What about if we rename to be more precise, for example `retrieve_row_by_key` or up to you if you have better candidate